cscs-checks/system/openstack/s3_check.py

```diff
@@ -9,135 +9,172 @@
 
 
 class S3apiCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self):
-        endpoint = 'object.cscs.ch'
-        self.descr = 'S3API check for (%s)' % endpoint
-        self.tags = {'ops', 'object_store'}
-        self.valid_systems = ['dom:gpu', 'daint:gpu']
-        self.valid_prog_environs = ['PrgEnv-gnu']
-        self.time_limit = '5m'
-        self.maintainers = ['VH', 'GLR']
-        self.executable = 's3_test.sh'
-        self.username = getpass.getuser()
-        if self.current_system.name in ['dom']:
+    endpoint = 'object.cscs.ch'
+    tags = {'ops', 'object_store'}
+    valid_systems = ['dom:gpu', 'daint:gpu']
+    valid_prog_environs = ['builtin']
+    time_limit = '5m'
+    executable = 's3_test.sh'
+    maintainers = ['VH', 'GLR']
+    username = getpass.getuser()
+
+    @run_after('init')
+    def add_production_tag(self):
+        if self.current_system.name in {'dom'}:
             self.tags |= {'production'}
 
+    @run_after('init')
+    def set_description(self):
+        self.descr = f'S3API check for ({self.endpoint})'
+
 
 @rfm.simple_test
 class S3apiCreateBucket(S3apiCheck):
-    def __init__(self):
-        super().__init__()
+    reference = {
+        '*': {
+            # below 1.1 secs is ok
+            'bucket_ctime': (1, None, 0.01, 's'),
+        }
+    }
+
+    @run_before('run')
```

Comment:
```suggestion
    @run_after('init')
```

Because it would show more information with the future `--describe` option.